Update Node version and other dev improvements

joninousthl opened the pull request:
- Update Node.js (and npm) in frontend-maven-plugin configuration to most recent LTS version. Using Node version newer than 7.3.0 enables use of NODE_EXTRA_CA_CERTS env variable which is needed when using a MITM proxy such as Zscaler.

- Update dev instructions in README to use Node, npm and other tools installed via frontend-maven-plugin instead of externally/globally installed ones. This improves stability of built since all developers will be using the same tools versions.

- Bower and Grunt specific commands are no longer mentioned in README's dev instructions directly. Instead, these will be run either through frontend-maven-plugin executions or package.json's scripts.

- Add mention of required tools (JDK 8 and Maven 3.8) in README.

- Remove dependencies Karma and Jasmine from project (package.json & gruntfile.js). These libraries are usually used for testing but termed-ui does not include any actual tests at the moment. There was basic "placeholder test" in test/concepts.test.js but this is now removed.

- Previously Karma was used with Grunt to keep the process alive when running "grunt dev". grunt-contrib-watch is now used for that purpose instead.

- Remove Angular and Angular CLI from package.json. Angular is installed via Bower. Angular CLI is not needed / does not work with Angular.js.

- Remove grunt-contrib-uglify since it was not being used.

- Commit package-lock.json into version control because it is recommended by npm documentation. With committed package-lock.json every developer should have same versions of dependencies (and their transitive dependencies) which improves project build stability.

Issue: TERMED-238

joninousthl committed:
Update Node version and other dev improvements

- Update Node.js (and npm) in frontend-maven-plugin configuration to most recent LTS version. Using Node version newer than 7.3.0 enables use of NODE_EXTRA_CA_CERTS env variable which is needed when using a MITM proxy such as Zscaler.

- Update dev instructions in README to use Node, npm and other tools installed via frontend-maven-plugin instead of externally/globally installed ones. This improves stability of built since all developers will be using the same tools versions.

- Bower and Grunt specific commands are no longer mentioned in README's dev instructions directly. Instead, these will be run either through frontend-maven-plugin executions or package.json's scripts.

- Add mention of required tools (JDK 8 and Maven 3.8) in README.

- Remove dependencies Karma and Jasmine from project (package.json & gruntfile.js). These libraries are usually used for testing but termed-ui does not include any actual tests at the moment. There was basic "placeholder test" in test/concepts.test.js but this is now removed.

- Previously Karma was used with Grunt to keep the process alive when running "grunt dev". grunt-contrib-watch is now used for that purpose instead.

- Remove Angular and Angular CLI from package.json. Angular is installed via Bower. Angular CLI is not needed / does not work with Angular.js.

- Remove grunt-contrib-uglify since it was not being used.

- Commit package-lock.json into version control because it is recommended by npm documentation. With committed package-lock.json every developer should have same versions of dependencies (and their transitive dependencies) which improves project build stability.

Issue: TERMED-238

soinonen commented:
Great general clean up and updates on the instructions. Tested it and works like a charm!

THL-mete-guneysel commented:
Tested with Windows environment and no problems. The npm scripts **start** and **build** need to be modified like this to work in Windows
```
.\\node\\node ...
```
This commit also simplified setting up the environment so good job 👍

joninousthl committed:
Update frontend-maven-plugin to latest version

Issue: TERMED-238

joninousthl committed:
Add Windows-specific run instructions to README

Issue: TERMED-238

joninousthl commented:
I added some Windows specific run instructions in README based on @mgusthl's comments. Also, I updated frontend-maven-plugin to latest that I forgot to do in the first commit.
